exp/clients/horizon/internal.go

```
@@ -13,6 +14,11 @@ func decodeResponse(resp *http.Response, object interface{}) (err error) {
 	defer resp.Body.Close()
 	decoder := json.NewDecoder(resp.Body)
 
+	// resp.Request should not be nil for Client requests
+	if resp.Request != nil {
```

Should we return an error if somehow it is `nil`?

---

When using mocks this is nil. So if we return error, the tests will fail.